### PR TITLE
fix: receive updated values from form elements

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Receive updated values from form elements

## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Form/Checkbox.tsx [hunk 1]
```diff
@@ -53,6 +53,10 @@ export const CheckboxWidget: React.FC<CheckBoxProps> = (props) => {
 
   const formInstance = Form.useFormInstance();
 
+  useEffect(() => {
+    setChecked(values?.initialValue as boolean);
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Checkbox.tsx:57]: Instead of casting, you can use the `Boolean` constructor which will actually convert it to a boolean.

File: packages/runtime/src/widgets/Form/Date/Date.tsx [hunk 2]
```diff
@@ -51,15 +55,23 @@ export const Date: React.FC<DateProps> = (props) => {
...
     } else {
       onChangeCallback(formattedDate);
     }
   };
 
+  useEffect(() => {
+    if (!value) {
+      setValue(
+        values?.initialValue || (dayjs(values?.value) as unknown as string),
+      );
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Date/Date.tsx:71]: Why would you not convert the initialDate also?

File: packages/runtime/src/widgets/Form/Date/Date.tsx [hunk 3]
```diff
@@ -51,15 +55,23 @@ export const Date: React.FC<DateProps> = (props) => {
   );
 
   const onDateChange = (date: unknown): void => {
+    setValue(date as string);
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Date/Date.tsx:58]: Just change the argument to type date?

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 4]
```diff
@@ -47,21 +47,28 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
...
     },
   );
 
+  useEffect(() => {
+    if (!selectedValues) {
+      const initVal = isString(values?.initialValue)
+        ? [values?.initialValue]
+        : values?.initialValue;
+      setSelectedValues(initVal as string[]);
+    }
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:69]: I think for this one `selectedValues` is the initial value.

Comment (sagardspeed2) [reply packages/runtime/src/widgets/Form/MultiSelect.tsx:69]: No, actually I did not assign the initial value in local state while state initiate, we are updating state value after initial value from the props are being evaluated by the `useRegisterBindings` only

Comment (evshi) [reply packages/runtime/src/widgets/Form/MultiSelect.tsx:69]: I meant that `initialValue` should be derived from `selectedValues` not the `value` prop.

Comment (sagardspeed2) [reply packages/runtime/src/widgets/Form/MultiSelect.tsx:69]: But on the first render, `selectedValues` will be empty array and we will assign the initialValues,  and when we change the any value of `Multiselect`, this time we only need to update the selectedValues, no need to reassign.

File: apps/kitchen-sink/src/ensemble/screens/forms.yaml [hunk 5]
```diff
@@ -122,13 +122,107 @@ View:
...
+                  id: myForm2
+                  styles:
+                    gap: 8px
+                    labelPosition: top
+                  onSubmit:
+                    executeCode:
+                      body: |
+                        const formattedVals = JSON.stringify(vals, null, 2)
+                        console.log(formattedVals)
+                        result.setText(formattedVals)
```
Comment (evshi) [inline apps/kitchen-sink/src/ensemble/screens/forms.yaml:146]: The result currently shows up in the other form.

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 6]
```diff
@@ -47,21 +47,28 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
...
       setSelectedValues,
       setOptions,
     },
   );
 
+  useEffect(() => {
+    if (!selectedValues) {
+      const initVal = isString(values?.initialValue)
+        ? [values?.initialValue]
+        : values?.initialValue;
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:67]: If this doesn't evaluate to an array, maybe we skip setting the initial value.

File: packages/runtime/src/widgets/Form/Date/Date.tsx [hunk 7]
```diff
@@ -50,16 +54,25 @@ export const Date: React.FC<DateProps> = (props) => {
...
       onChangeCallback(formattedDate);
     }
   };
 
+  useEffect(() => {
+    if (!value) {
+      setValue(
+        values?.initialValue ||
+          (dayjs(values?.initialValue) as unknown as string),
+      );
```
[the remark this answers is not in the document]
Comment (sagardspeed2) [reply packages/runtime/src/widgets/Form/Date/Date.tsx:72]: Yes

File: packages/runtime/src/widgets/Form/Date/Date.tsx [hunk 8]
```diff
@@ -84,7 +94,7 @@ export const Date: React.FC<DateProps> = (props) => {
         onChange={onDateChange}
         placeholder={values?.hintText}
         style={{ width: "100%", ...values?.styles }}
-        value={dayjs(values?.value)}
+        value={String(dayjs(values?.value))}
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Date/Date.tsx:97]: Why is this change needed?

**Review by evshi — APPROVED** [review]
